state-chain/chains/src/address.rs

```diff
@@ -68,6 +68,28 @@ pub enum EncodedAddress {
 	Btc(Vec<u8>),
 }
 
+impl Serialize for EncodedAddress {
+	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+	where
+		S: serde::Serializer,
+	{
+		match self {
+			EncodedAddress::Btc(addr) => format!("0x{}", hex::encode(addr)).serialize(serializer),
+			EncodedAddress::Dot(addr) => format!("0x{}", hex::encode(addr)).serialize(serializer),
+			EncodedAddress::Eth(addr) => format!("0x{}", hex::encode(addr)).serialize(serializer),
+		}
+	}
+}
+
+impl<'de> Deserialize<'de> for EncodedAddress {
+	fn deserialize<D>(_deserializer: D) -> Result<Self, D::Error>
+	where
+		D: serde::Deserializer<'de>,
+	{
+		unimplemented!()
```

Comment:
I would rather we didn't have this, seems a bit dangerous. 

The implementation is just `clean_foreign_chain_address` from the api crate, but refactoring and moving all this is beyond the scope of this PR. 

I'll make the change and then we can merge.

Reply:
thanks